inc/notificationtargetcommonitilobject.class.php

```diff
@@ -1319,13 +1319,12 @@ function getDataForObject(CommonDBTM $item, array $options, $simple = false) {
             $tmp['##followup.isprivate##']   = Dropdown::getYesNo($followup['is_private']);
 
             // Check if the author need to be anonymized
-            if (ITILFollowup::getById($followup['id'])->isFromSupportAgent()
-               && $is_self_service
+            if ($is_self_service && $need_anonymize
+               && ITILFollowup::getById($followup['id'])->isFromSupportAgent()
                && !empty($anon_name = User::getAnonymizedName(
```

Comment:
What is the benefit of adding `$need_anonymize` to the condition here ?
Isn't it doing the exact same thing with a duplicated condition ?

`$need_anonymize` is false if Entity::ANONYMIZE_DISABLED
`!empty($anon_name = User::getAnonymizedName))` is also false if Entity::ANONYMIZE_DISABLED

Reply:
fixed, useless check removed